test/e2e/singlecluster/pod_test.go

```diff
@@ -489,27 +490,28 @@ var _ = ginkgo.Describe("Pod groups", func() {
 				})
 			})
 
-			replacementPods := make(map[types.NamespacedName]types.NamespacedName, len(defaultPriorityGroup))
-			ginkgo.By("Create replacement pods as soon as the default-priority pods are Failed", func() {
-				gomega.Eventually(func(g gomega.Gomega) int {
+			ginkgo.By("Wait for default-priority pods are Failed", func() {
```

Comment:
Why do we split this step into two? 

It was intentionally designed to create replacement pods as soon as failures are observe to simulate what user supplied controller would do in case of a failure. 

I'm ok with the change if really needed. If we can make the test stable just by extending timeout, then this would be my preference.

Reply:
I have synced with @mbobrovskyi and the underlying issue is that as soon as the replacement pod is created the pod_controller removes the finalizer for the other Pod (see [here](https://github.com/kubernetes-sigs/kueue/pull/2731#discussion_r1699688583)).

So, occasionally `g.Expect(k8sClient.Get(ctx, origKey, &p)).To(gomega.Succeed())` may fail for the second pod due to 404.

Splitting into two phases eliminates the problem, because we don't create the replacement until we checked both pods are failed. Alternative approach, would be to check Failed OR 404, but this is an extra complication.

I'm ok to refactor the test to simplify.